src/fivetran/lambdaOpsGenie/lambdaHandler.ts

```diff
@@ -1,27 +1,40 @@
-import { FiveTranRequest, Secrets } from '../../commands/fiveTranInterface'
+import { FiveTranRequest, Secrets, State } from '../../commands/fiveTranInterface'
 import { getOpsGenieAlertsFromPoint, getAllOpsGenieAlerts } from '../../commands/opsGenieCommands'
 import { secureEnvironmentVariable } from '../../aws/secretsClient'
 import { secret } from './fiveTranSecret'
 
 let opsGenieApiKey: string
 
 export async function handler(request: FiveTranRequest) {
+  // Verify Secret
   fiveTranSecretVerification(request.secrets)
-  return await update(request.state.lastUpdate)
+
+  return await update(request.state)
 }
-async function update(state: string) {
+async function update(state: State) {
+  // opsGenie secret Decrypt
   const decryptRequest: Promise<void> = decryptSlackSecret()
   await decryptRequest
   opsGenieApiKey = process.env.OPS_GENIE_SECRET || ''
 
+  // Pagination Intitalise
+  let paginationCounter = state?.paginationCounter || 0
+
   // Fetch records using api calls
-  const insertTransactions = await apiResponse(state)
+  const { insertTransactions, hasMore } = await apiResponse(state.lastUpdate, paginationCounter)
+
+  // Increment pagination counter if hasMore is true
+  hasMore ? (paginationCounter += 1) : (paginationCounter = 0)
```

Comment:
format condition ? (TRUE BLOCK) : (FALSE BLOCK)

Reply:
```suggestion
paginationCounter = hasMore ? paginationCounter + 1 : 0
```

You currently have assignment occuring within your ternary operator - this is uncommon which in turn harms readability.

Prefer explicit assignment